operator/v1/types_console.go

```diff
@@ -106,6 +106,38 @@ type ConsoleCustomization struct {
 	// SVG format preferred
 	// +optional
 	CustomLogoFile configv1.ConfigMapFileReference `json:"customLogoFile,omitempty"`
+	// developerCatalog allows to configure the shown developer catalog categories.
+	// +optional
+	DeveloperCatalog DeveloperConsoleCatalogCustomization `json:"developerCatalog,omitempty"`
+}
+
+// DeveloperConsoleCatalogCustomization allow the cluster admin to configure developer catalog.
+type DeveloperConsoleCatalogCustomization struct {
+	// categories which are shown the in the devconsole.
+	// +optional
+	Categories []DeveloperConsoleCatalogCategory `json:"categories,omitempty"`
+}
+
+// DeveloperConsoleCatalogCategoryMeta defines a category incl. label and filter tags
+// for the developer console catalog.
+type DeveloperConsoleCatalogCategoryMeta struct {
+	// ID is an identifier used in the URL to enable deep linking in console.
+	ID string `json:"id"`
+	// label defines a category display label.
+	Label string `json:"label"`
+	// tags in a list of strings that will match the category.
+	// +optional
+	Tags []string `json:"tags,omitempty"`
+}
+
+// DeveloperConsoleCatalogCategory defines a top level category incl. label, filter tags
```

Comment:
The comment here should match the API property instead of the name of the struct since this will be used for the CRD property description. I'd avoid the incl. abbreviation here since this is publicly facing doc.

Reply:
@jhadvig Changed type messages to:

for DeveloperConsoleCatalogCustomization:

> // developerCatalog allow the cluster admin to configure developer catalog.

for DeveloperConsoleCatalogCategoryMeta and DeveloperConsoleCatalogCategory:

> // category for the developer console catalog.